src/Integration.Vsix/Helpers/IStatusBarNotifier.cs

```diff
@@ -35,20 +36,24 @@ internal interface IStatusBarNotifier
     [PartCreationPolicy(CreationPolicy.Shared)]
     internal class StatusBarNotifier : IStatusBarNotifier
     {
-        private IVsStatusbar vsStatusBar;
+        private readonly IVsUIServiceOperation vSServiceOperation;
+        private readonly IThreadHandling threadHandling;
 
         [ImportingConstructor]
-        public StatusBarNotifier([Import(typeof(SVsServiceProvider))] IServiceProvider serviceProvider)
+        public StatusBarNotifier(IVsUIServiceOperation vSServiceOperation, IThreadHandling threadHandling)
         {
-            RunOnUIThread.Run(() =>
-            {
-                vsStatusBar = serviceProvider.GetService(typeof(IVsStatusbar)) as IVsStatusbar;
-            });
+            this.vSServiceOperation = vSServiceOperation;
+            this.threadHandling = threadHandling;
         }
 
         public void Notify(string message, bool showSpinner)
         {
-            RunOnUIThread.Run(() =>
+            vSServiceOperation.Execute<IVsStatusbar, IVsStatusbar>(vsStatusBar => DoNotify(vsStatusBar, message, showSpinner));
+        }
+
+        private void DoNotify(IVsStatusbar vsStatusBar, string message, bool showSpinner)
+        {
+            threadHandling.RunOnUIThread(() =>
```

Comment:
Is `threadHandling.RunOnUIThread` required here? The service operation will already have switched to the UI thread.